treex/metrics/mean_absolute_error.py

```diff
@@ -0,0 +1,85 @@
+import typing as tp
+
+import jax.numpy as jnp
+
+from treex import types
+from treex.metrics.mean import Mean
+
+
+def _mean_absolute_error(y_pred: jnp.ndarray, y_true: jnp.ndarray) -> jnp.ndarray:
+    """Calculates values required to update/compute Mean Absolute Error. Cast y_pred to have the same type as y_true.
+
+    Args:
+        y_pred: Predicted tensor
+        y_true: Ground truth tensor
+
+    Returns:
+        jnp.ndarray values needed to update Mean Absolute Error
+    """
+
+    y_true = y_true.astype(y_pred.dtype)
+    return jnp.mean(jnp.abs(y_pred - y_true), axis=-1)
```

Comment:
Curious if `jnp.mean` here is actually necessary.

Reply:
@cgarciae yes its not necessary sin it is inheriting from Mean.